### PR TITLE
Update Microsoft.Data.Sqlite packages to version 2.2.4.0

### LINKED ISSUE #273: Upgrade Microsoft.Data.Sqlite to prevent System.InvalidOperationException: SqliteConnection does not support nested transactions.
The issue described here:

https://github.com/aspnet/Microsoft.Data.Sqlite/issues/532

could be causing the occasional SQLite related exception we've been seeing:

```
System.InvalidOperationException: SqliteConnection does not support nested transactions.
   at Microsoft.Data.Sqlite.SqliteConnection.BeginTransaction(IsolationLevel isolationLevel)
   at GVFS.Virtualization.BlobSize.BlobSizes.BlobSizesDatabaseWriter.TryAddSizes(ConcurrentQueue`1 sizes, Int32& errorCode, String& error)
   at GVFS.Virtualization.BlobSize.BlobSizes.FlushDbThreadMain()\",\"ErrorMessage\":\"FlushDbThreadMain caught unhandled exception, exiting process\"}"
```

If there is an internal rollback without the [fix mentioned above](https://github.com/aspnet/Microsoft.Data.Sqlite/issues/532) `SqliteTransaction.Complete()` will *not* be called leaving a non-null `Transaction` set on `BlobSizesDatabaseWriter.connection`

We should pick up the fix and see if the issue persists.

### PR DESCRIPTION
**Pull request by kewillford**
Updating the Sqlite packages to the latest

Fixes #273

### CODE REVIEW

File: GVFS/GVFS/GVFS.Windows.csproj [hunk 1]
```diff
@@ -186,9 +178,17 @@ xcopy /Y $(BuildOutputDir)\GitHooksLoader\bin\$(Platform)\$(Configuration)\GitHo
...
-  <Import Project="..\..\..\packages\SQLitePCLRaw.lib.e_sqlite3.osx.1.1.7\build\net35\SQLitePCLRaw.lib.e_sqlite3.osx.targets" Condition="Exists('..\..\..\packages\SQLitePCLRaw.lib.e_sqlite3.osx.1.1.7\build\net35\SQLitePCLRaw.lib.e_sqlite3.osx.targets')" />
-  <Import Project="..\..\..\packages\SQLitePCLRaw.lib.e_sqlite3.v110_xp.1.1.7\build\net35\SQLitePCLRaw.lib.e_sqlite3.v110_xp.targets" Condition="Exists('..\..\..\packages\SQLitePCLRaw.lib.e_sqlite3.v110_xp.1.1.7\build\net35\SQLitePCLRaw.lib.e_sqlite3.v110_xp.targets')" />
+  <Import Project="..\..\..\packages\SQLitePCLRaw.lib.e_sqlite3.linux.1.1.12\build\net35\SQLitePCLRaw.lib.e_sqlite3.linux.targets" Condition="Exists('..\..\..\packages\SQLitePCLRaw.lib.e_sqlite3.linux.1.1.12\build\net35\SQLitePCLRaw.lib.e_sqlite3.linux.targets')" />
+  <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
+    <PropertyGroup>
+      <ErrorText>This project references NuGet package(s) that are missing on this computer. Use NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is {0}.</ErrorText>
```
Comment (wilbaker) [inline GVFS/GVFS/GVFS.Windows.csproj]: Was this added automatically by VS?

**Comment by wilbaker** [comment]
I saw the SQLite repo moved, out of curiosity, did you need to do anything special to pull this new version of SQLite?  Or did the nuget feed not change?

File: GVFS/GVFS/GVFS.Windows.csproj (hunk 1 above)
Comment (kewillford) [reply GVFS/GVFS/GVFS.Windows.csproj]: Yep because all I did was update the nuget packages for the solution.  Do you think we need to remove it?

Comment (wilbaker) [reply GVFS/GVFS/GVFS.Windows.csproj]: I think it looks fine, I was just wondering if you had to add it by hand.

**Comment by kewillford** [comment]
@wilbaker All I did was go to the manage NuGet packages for solution and update the Microsoft.Data.Sqlite package.  Everything else was VS and NuGet doing whatever they do.

**Comment by wilbaker** [comment]
Have you tested that the older version of SQLite we've been using can read/write to databases created by this version? (I would hope the answer is that it can 😄 but if it can't we'll need to bump the disk layout version)

**Comment by kewillford** [comment]
@wilbaker tested both reading an old sqlite database with upgraded packages and reading a new sqlite database with the old packages and everything worked without issue.  Thanks!

**Pull request merged**